### PR TITLE
[Feature] Add more comprehensive error handling

### PR DESCRIPTION
**Pull request by HaiyiMei**
- fix typo
- add more `assert` to prevent improper input
- add version control for plugins of `XRFeitoriaBpy` and `XRFeitoriaUnreal`
- add a threading of checking if the engine is alive. If the engine dead detected, throw an error with engine output

### CODE REVIEW

File: xrfeitoria/utils/runner.py [hunk 1]
```diff
@@ -117,10 +117,28 @@ def __init__(
         if not self.engine_exec.exists() or not self.engine_exec.is_file():
             raise FileExistsError(f'Engine executable not valid: {self.engine_exec}')
         logger.info(f'Using engine executable: "{self.engine_exec.as_posix()}"')
-        self.project_path = Path(project_path).resolve() if project_path else None
+        if project_path:
+            self.project_path = Path(project_path).resolve()
+            assert self.project_path.exists(), (
+                f'Project path not valid: "{self.project_path.as_posix()}"\n'
+                'Please check in `xf.init_blender(project_path=...)` or `xf.init_unreal(project_path=...)`'
```
Comment (yl-1993) [inline xrfeitoria/utils/runner.py:124]: A few typos:
- "Project path `is` not valid" or "Project path does not exist"
- Please check `xxx`

Comment (HaiyiMei) [reply xrfeitoria/utils/runner.py:124]: resolved

Comment (HaiyiMei) [reply xrfeitoria/utils/runner.py]: fixed

**Comment by HaiyiMei** [comment]
@yl-1993 This PR is ready to be merged.

**Review by yl-1993 — APPROVED** [review]

**Pull request merged**